js_modules/dagit/packages/core/src/assets/OverdueTag.tsx

```diff
@@ -0,0 +1,227 @@
+  const result = useQuery<OverduePopoverQuery, OverduePopoverQueryVariables>(
+    OVERDUE_POPOVER_QUERY,
+    {variables: {assetKey: {path: assetKey.path}, timestamp}},
+  );
+
+  const data =
+    result.data?.assetNodeOrError.__typename === 'AssetNode' ? result.data.assetNodeOrError : null;
+
+  if (!data) {
+    return (
+      <Box style={{width: 600}}>
+        <LoadingSpinner purpose="section" />
+      </Box>
+    );
+  }
+
+  if (!data.freshnessPolicy) {
+    return <Box style={{width: 600}}>No freshness policy or timestamp.</Box>;
+  }
+
+  const minutesLate = humanizedMinutesLateString(liveData.freshnessInfo?.currentMinutesLate || 0);
+
+  return (
+    <Box style={{width: 600}}>
+      <Box padding={12} border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}>
+        {`The latest materialization is derived from source data that is ${minutesLate} old. The asset's freshness policy requires it to be derived from data ${freshnessPolicyDescription(
+          data.freshnessPolicy,
+          'short',
+        )}`}
+      </Box>
+      <Box
+        padding={12}
+        style={{fontWeight: 600}}
+        border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}
+      >
+        Latest materialization:
+      </Box>
+      <Box
+        padding={12}
+        flex={{justifyContent: 'space-between'}}
+        border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}
+      >
+        <Timestamp timestamp={{ms: Number(timestamp)}} />
+        <TimeSinceWithOverdueColor
+          timestamp={Number(timestamp)}
+          maximumLagMinutes={data.freshnessPolicy.maximumLagMinutes}
+        />
+      </Box>
+      <Box padding={12} style={{fontWeight: 600}}>
+        Latest materialization sources data from:
+      </Box>
+      <Box
+        style={{maxHeight: '240px', overflowY: 'auto', marginLeft: -1, marginRight: -1}}
+        onClick={(e) => e.stopPropagation()}
+      >
+        <AssetMaterializationUpstreamTable
+          data={data}
+          timestamp={timestamp}
+          timestampDiffStyle="since-now"
+          assetKey={assetKey}
+        />
+      </Box>
+    </Box>
+  );
+};
+
+export const freshnessPolicyDescription = (
+  freshnessPolicy: FreshnessPolicy | null,
+  format: 'long' | 'short' = 'long',
+) => {
+  if (!freshnessPolicy) {
+    return '';
+  }
+
+  const {cronSchedule, maximumLagMinutes, cronScheduleTimezone} = freshnessPolicy;
+  const nbsp = '\xa0';
+  const cronDesc = cronSchedule
+    ? humanCronString(cronSchedule, cronScheduleTimezone ? cronScheduleTimezone : 'UTC').replace(
+        /^At /,
+        '',
+      )
+    : '';
+  const lagDesc =
+    maximumLagMinutes % 30 === 0
+      ? `${maximumLagMinutes / 60} hour${maximumLagMinutes / 60 !== 1 ? 's' : ''}`
+      : `${maximumLagMinutes} min`;
+
+  if (format === 'short') {
+    if (cronDesc) {
+      return `no more than ${lagDesc} old by${nbsp}${cronDesc}.`;
+    } else {
+      return `no more than ${lagDesc} old at any${nbsp}time.`;
```

Comment:
Ahh this is a bit awkward but it's to make sure that the popover never wraps just the word `time` to the last line since the description varies in length. There might be a more generic way we could do this for all tooltips actually 🤔